sample_tracking/app/login/signup.tsx

```diff
@@ -95,7 +105,41 @@ export default function SignUp(props: SignUpProps) {
         const name = `${formData.firstName} ${formData.lastName}`;
         const labValue = orgName ? availableOrgs[orgName] : '';
 
-        await createUserWithEmailAndPassword(auth, formData.email, formData.password);
+        await createUserWithEmailAndPassword(auth, formData.email, formData.password).catch((error) => {
+            console.log(error);
+            switch (error.code) {
+                case 'auth/weak-password':
+                    setErrorText({
+                        ...errorText,
+                        password: error.message.substring(error.message.indexOf(':') + 2, error.message.indexOf('('))
```

Comment:
its not very robust to search the error string for : and ( - what if the error format changes?  Can we just show the message verbatim?  Its more stable, albeit less pretty.